notebook/services/contents/tests/test_contents_api.py

```diff
@@ -518,7 +518,8 @@ def test_delete_dirs(self):
         for name in sorted(self.dirs + ['/'], key=len, reverse=True):
             listing = self.api.list(name).json()['content']
             for model in listing:
-                self.api.delete(model['path'])
+                if os.path.exists(model['path']):
```

Comment:
API paths shouldn't be used as filesystem paths - you can't pass them to functions like `os.path.exists()`, because they probably don't start from the same directory where the tests are running. This check shouldn't be needed in any case.